### PR TITLE
Add colors in help outputs of Airfow CLI commands #28789

### LINKED ISSUE #28789: Add colors in help outputs of Airfow CLI commands
### Body

Folowing up after https://github.com/apache/airflow/pull/22613#issuecomment-1374530689 - seems that there is a new [rich-argparse](https://github.com/hamdanal/rich-argparse) project that might give us the option without rewriting Airflow's argument parsing to click (click has a number of possible performance issues that might impact airlfow's speed of CLI command parsing) 

Seems this might be rather easy thing to do (just adding the formatter class for argparse).

Would be nice if someone implements it and tests (also for performance of CLI).

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by loulsb**
closes: #28789

@potiuk I started with changing formatter_class as described here https://github.com/hamdanal/rich-argparse#usage. I get the following output:

airflow --help
Screenshot 2023-01-23 225405

airflow users --help
Screenshot 2023-01-23 225428

airflow users add-role --help
Screenshot 2023-01-23 225502

In the issue you said "Would be nice if someone implements it and tests (also for performance of CLI)". Any idea how I can do that?

### CODE REVIEW

**Comment by potiuk** [comment]
Looks cool. You need to rebase, and fix the static checks and CLI tests (the SSM test errors will likely go away after rebase - it was broken main I think).

**Comment by potiuk** [comment]
I think performance in this case is very little impacted. The Click change was much more of the problem because click works with python-decorated methods and it has natural tendency to importing much more and the imports would take much more time.

**Comment by potiuk** [comment]
> @potiuk what should I do next?

Wait until we approve the tests (as you are first time contributor) and somoene reviews it/merges it. I just run the tests. Let's see. If you see your PR is not picked up, just gently (as you did) ping in the PR so that someone will take a look. We have ~ 100 PRs to take a look, review, comment and people who are doing it are mostly doing it in their free/voluntary time (this is how OSS works) and they have other things to do. So being patient is important when contributing. We explain it in a few places : https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#id61 for one.

**Review by potiuk — APPROVED** [review]
Tested it locally and it looks cool :). Anyone has any comments here? @blag @ashb I think you've been looking at click before and this change **just** gives us some nicer colors and makes things look more modern, while not compromising the CLI speed and does not require heavy modifications. It uses rich under-the-hood so it will also nicely perform auto-detction of terminal capabilities and the like. 

I cannot think of a negative consequences of the change and airflow's CLI help output will be quite a bit more readable for most people.

**Review by Taragolis — APPROVED** [review]
Looks really nice

<details>
  <summary>Solarized Dark</summary>

image
  
</details>

**Comment by potiuk** [comment]
> * Wait until we approve the tests (as you are first time contributor) and somoene reviews it/merges it.
> 
> I am sorry I wasn't clear. I meant what should I do about the test that failed last time it ran. I thank you for your guidance. I asked because your comment earlier (the SSM test errors will likely go away after rebase) helped me understand why I had a failing test before and I don't know how to run a test manually (now I understand that I can't because I am a first time contributor). Thanks again for your help.

No worries :). All good now. Tests are green. Let's see if there are other comments and we can merge that one :)

File: airflow/cli/cli_parser.py [hunk 1]
```diff
@@ -2201,46 +2202,56 @@ def _remove_dag_id_opt(command: ActionCommand):
...
-class AirflowHelpFormatter(argparse.HelpFormatter):
+class AirflowHelpFormatter(RichHelpFormatter):
     """
     Custom help formatter to display help message.
 
     It displays simple commands and groups of commands in separate sections.
     """
 
-    def _format_action(self, action: Action):
+    def _rich_format_action(self, action: Action):
```
Comment (uranusjr) [inline airflow/cli/cli_parser.py]: Is there guarantee this would not break with newer `rich-argparse` versions? It would be best if we could get some input from the package’s developers.

Comment (loulsb) [reply airflow/cli/cli_parser.py]: That is a good question.
@hamdanal could we bother you in having your opinion if using _rich_format_action is safe?

Comment (hamdanal) [reply airflow/cli/cli_parser.py]: Awesome to see `rich-argparse` work in such an amazing project. Please let me know if you find any bugs or quirks, I'll be happy to address them.

The TLDR answer to your question is, I really doubt the interface of `_rich_format_action` will change a lot but it is a private method so stability cannot be promised.

### Long answer
`_rich_format_action` and some other `RichHelpFormatter` methods with similar names are based on the same methods in `argparse.HelpFormatter` without the `_rich` prefix. Since the `argparse` methods are private, the equivalent `_rich...` methods are also private. Have the original methods been public, I would've made the `_rich` methods public. The point is I have no intention or interest in introducing a breaking change in these methods however if the interfaces of the original `argparse` methods change, the `_rich` methods will follow that change. I think this is a very unlikely scenario because `argparse` is currently maintained in this "extremely stable mode" and rarely gets any updates (even bug fixes take years to be accepted if ever). That said I don't like to give you a promise I can't honor, hence the short answer above.

### Possible solution
From what I understand reading the implementation of this method here, you seem to be interested in two things (please correct me if I am wrong):
1. Sort the subcommands into two sections: Groups and Commands
2. Print a section title at the top of each section

Both of these are possible using the vanilla `argparse.HelpFormatter` method [`_iter_indented_subactions`](https://github.com/python/cpython/blob/main/Lib/argparse.py#L635-L643) without needing to change any `rich-argparse` specific code. The trick is to create an empty "fake" action with `nargs=0` that prints the section header before its content. This way you get something that works with both the original formatter and the `rich-argparse` formatter. I think this will be even simpler implementation compared to your current solution.

Comment (loulsb) [reply airflow/cli/cli_parser.py]: Thank you @hamdanal. I'll try your solution

Comment (loulsb) [reply airflow/cli/cli_parser.py]: _iter_indented_subactions seems to work. I get the same result as _format_action before

**Review by uranusjr — APPROVED** [review]
